### PR TITLE
Use a backoff approach when connecting to database

### LINKED ISSUE #159: Add wait/backoff for postgres
It would be great if Clair waited for Postgres to become available rather than crashing out. IMO, this should be done in the Go code itself. See https://medium.com/@kelseyhightower/12-fractured-apps-1080c73d481c#.b9pu0hu5u for more details.

In the meantime, I hacked together: https://github.com/amouat/clair-with-wait in case anyone finds it useful.

### PR DESCRIPTION
**Pull request by ajgreenb**
I followed the implementation detailed in [12 Fractured Apps](https://medium.com/@kelseyhightower/12-fractured-apps-1080c73d481c#b9d9) (mentioned in #159). I guess this would also supersede #252.

Closes #159

### CODE REVIEW

File: cmd/clair/main.go [hunk 1]
```diff
@@ -132,10 +132,22 @@ func Boot(config *Config) {
 	st := stopper.NewStopper()
 
 	// Open database
-	db, err := database.Open(config.Database)
-	if err != nil {
-		log.Fatal(err)
+	var db database.Datastore
+	var dbError error
+	maxConnectionAttempts := 20
```
Comment (jzelinskie) [inline cmd/clair/main.go]: can we make this a constant at the top of the file and rename it `maxDBConnectionAttempts`?

File: cmd/clair/main.go [hunk 2]
```diff
@@ -132,10 +132,22 @@ func Boot(config *Config) {
...
+	for attempts := 1; attempts <= maxConnectionAttempts; attempts++ {
+		db, dbError = database.Open(config.Database)
+		if dbError == nil {
+			break
+		}
+		log.Error(dbError);
+		time.Sleep(time.Duration(attempts) * time.Second)
 	}
+
```
Comment (jzelinskie) [inline cmd/clair/main.go]: no need for this new line

File: cmd/clair/main.go [hunk 3]
```diff
@@ -132,10 +132,22 @@ func Boot(config *Config) {
...
-		log.Fatal(err)
+	var db database.Datastore
+	var dbError error
+	maxConnectionAttempts := 20
+	for attempts := 1; attempts <= maxConnectionAttempts; attempts++ {
+		db, dbError = database.Open(config.Database)
+		if dbError == nil {
+			break
+		}
+		log.Error(dbError);
```
Comment (jzelinskie) [inline cmd/clair/main.go]:
```go
log.WithError(dbError).Error("failed to start connect to database")
```

**Review by jzelinskie — COMMENTED** [review]
Sorry for the long wait for a review. This looks really useful!

**Comment by ajgreenb** [comment]
@jzelinskie thanks for the review! Changes pushed.

**Review by jzelinskie — APPROVED** [review]
LGTM! Thanks for contributing!

**Comment by ajgreenb** [comment]
Thanks! 🎉

**Pull request merged**